### PR TITLE
Stabilize flaky test 25374

### PR DESCRIPTION
**Pull request by filiphric**
### Description

Stabilizes some flaky tests connected to issue #25374. The flakiness happened because backend sometimes sends `last_used_param_values` which causes an immediate redirect. This caused some test to throw an error:

```diff
Timed out retrying after 4000ms
- actual
+ expected 

-'?equal_to=1%2C2%2C3'
+'?equal_to='
```

This was fixed by stubbinh the `last_used_param_values` parameter as shown here: https://github.com/metabase/metabase/blob/5f4df4075c6ed622a0935baa770faddd4645daff/e2e/support/helpers/e2e-misc-helpers.js#L224-L234

The other part of the fix is to wait for these stubbed responses to arrive before proceeding to the next step of the test. Since the intercept is dynamic (are there any good reasons for that though? cc: @nemanjaglumac) I needed to create a custom command that would accept regex and wait for it. see the [`e2e/support/commands/utils/waitAlias.ts`](https://github.com/metabase/metabase/blob/5f4df4075c6ed622a0935baa770faddd4645daff/e2e/support/commands/utils/waitAlias.ts) file.

I removed the `@flaky` tag from these tests which messed up the diff. I tried to prevent this, but it seems our linters have a different opinion on indenting.

### How to verify

Test burn-in -> https://github.com/metabase/metabase/actions/runs/11601385733

### CODE REVIEW

File: e2e/support/commands/utils/waitAlias.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      pattern.test(alias),
+    );
+
+    if (matchingAliases.length === 0) {
+      throw new Error(`No aliases found matching pattern: ${pattern}`);
+    }
+
+    return cy.wait("@" + matchingAliases[0]);
+  });
+});
```
Comment (iethree) [inline e2e/support/commands/utils/waitAlias.ts:29]: snazzy!

**Review by iethree — APPROVED** [review]

File: e2e/support/helpers/e2e-misc-helpers.js [hunk 2]
```diff
@@ -221,7 +221,17 @@ function visitDashboardById(dashboard_id, config) {
...
+    cy.intercept(
+      {
+        method: "GET",
+        url: `/api/dashboard/${dashboard_id}*`,
+      },
+      req => {
+        req.reply(res => {
+          res.body.last_used_param_values = {}; // remove param values to prevent random redirects caused by cached redirect
+        });
+      },
```
Comment (iethree) [inline e2e/support/helpers/e2e-misc-helpers.js:233]: I think this may have broken some other tests? 🤔

Comment (filiphric) [reply e2e/support/helpers/e2e-misc-helpers.js:233]: did it? I was wondering how do we even match this intercept. it does not seem like we wait for it in any tests, we just match this request for no apparent reason. I had to create that waitAlias utility to match this intercept

File: e2e/support/commands/utils/waitAlias.ts (hunk 1 above)
Comment (filiphric) [reply e2e/support/commands/utils/waitAlias.ts:29]: Ideally I’d like this to be a cy.wait() override in the future, but didn’t want to get too deep into this

File: e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js [hunk 3]
```diff
@@ -1782,14 +1782,22 @@ describe("issue 25374", () => {
...
-    );
+    // we are overriding the intercept set in e2e/support/helpers/e2e-misc-helpers.js
+    cy.intercept(
+      {
+        method: "GET",
+        url: "/api/dashboard/*?dashboard_load_id=**",
+      },
+      req => {
+        req.reply(res => {
+          res.body.last_used_param_values = {}; // remove param values to prevent random redirects caused by cached redirect
```
Comment (uladzimirdev) [inline e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js:1793]: we shouldn't override server behaviour in tests, should we? even if it's a flaky test, such change only hides the problem

File: e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js [hunk 4]
```diff
@@ -1782,14 +1782,22 @@ describe("issue 25374", () => {
   };
 
   beforeEach(() => {
-    cy.intercept("POST", "/api/card/*/query").as("cardQuery");
-    cy.intercept("POST", "/api/dashboard/*/dashcard/*/card//*/query").as(
-      "dashcardQuery",
-    );
+    // we are overriding the intercept set in e2e/support/helpers/e2e-misc-helpers.js
+    cy.intercept(
```
Comment (uladzimirdev) [inline e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js:1786]: afaik we add a new intercept here, there is no way to override old intercepts, but only last matching intercept will actually handle request, so idk if "overriding" is a correct word here

File: e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js (hunk 3 above)
Comment (kamilmielnik) [reply e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js:1792]: For the record the problem is that `last_used_param_values` are updated asynchronously in the backend which makes these kind of tests non-deterministic.
The solution is to use `changeSynchronousBatchUpdateSetting` util, see #49379.
Related Slack conversation: https://metaboat.slack.com/archives/C010L1Z4F9S/p1727947712609579

**Pull request closed without merging**